File: languages/go/entity/identityV2/decimal_attribute_descriptor.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// DecimalAttributeDescriptor represents a descriptor for an attribute with a decimal value.
type DecimalAttributeDescriptor struct {
	AttributeDescriptor
}

// NewDecimalAttributeDescriptor creates a new DecimalAttributeDescriptor instance with the provided details.
func NewDecimalAttributeDescriptor(name string, id string, description string, required bool) DecimalAttributeDescriptor {
	a := AttributeDescriptor{
		DisplayName: name,
		Id:          id,
		Description: description,
		Required:    required,
	}
	return DecimalAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func NewDecimalAttributeDescriptorFromProto(s *proto.DecimalAttributeDefinitionV2) DecimalAttributeDescriptor {
	if s == nil {
		return DecimalAttributeDescriptor{}
	}
	a := AttributeDescriptor{
		DisplayName: s.GetId(),
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
	return DecimalAttributeDescriptor{
		AttributeDescriptor: a,
	}
}

func (s DecimalAttributeDescriptor) ToProto() *proto.DecimalAttributeDefinitionV2 {
	return &proto.DecimalAttributeDefinitionV2{
		DisplayName: s.DisplayName,
		Id:          s.Id,
		Description: s.Description,
		Required:    s.Required,
	}
}

File: languages/python/bloock/entity/identity_v2/decimal_attribute_descriptor.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto
from bloock.entity.identity_v2.attribute_descriptor import AttributeDescriptor


class DecimalAttributeDescriptor(AttributeDescriptor):
    @staticmethod
    def from_proto(a: proto.DecimalAttributeDefinitionV2) -> DecimalAttributeDescriptor:
        return DecimalAttributeDescriptor(
            display_name=a.display_name,
            technical_name=a.id,
            description=a.description,
            required=a.required,
        )

    def to_proto(self) -> proto.DecimalAttributeDefinitionV2:
        return proto.DecimalAttributeDefinitionV2(
            display_name=self.display_name,
            id=self.technical_name,
            description=self.description,
            required=self.required,
        )